### PR TITLE
[PLAT-7855] Update to latest bugsnag-android

### PR DESCRIPTION
**Pull request by kstenerud**
## Goal

This is the minimal change set to get the bugsnag-cocos2dx plugin working with the latest bugsnag-android.

Note: The example app has not been updated, nor has any documentation. These will be done in a separate PR.

## Testing

Tested manually in a cocos2d-x app to ensure that it can load, init, and report crashes.

### CODE REVIEW

File: src/android/bugsnag-plugin-android-cocos2dx/src/main/java/com/bugsnag/android/BugsnagCocos2dxPlugin.java [hunk 1]
```diff
@@ -1,90 +1,33 @@
...
 
-    private void configureNotifierDetails() {
-        Notifier notifier = Notifier.getInstance();
+        Notifier notifier = client.getNotifier();
         notifier.setName("Bugsnag Cocos2d-x");
         notifier.setVersion(pluginVersion);
-        notifier.setURL("https://github.com/bugsnag/bugsnag-cocos2dx");
-    }
+        notifier.setUrl("https://github.com/bugsnag/bugsnag-cocos2dx");
+        notifier.setDependencies(Arrays.asList(new Notifier())); // depend on bugsnag-android
```
Comment (lemnik) [inline src/android/bugsnag-plugin-android-cocos2dx/src/main/java/com/bugsnag/android/BugsnagCocos2dxPlugin.java]: nit:
```suggestion
        notifier.setDependencies(Collections.singletonList(new Notifier())); // depend on bugsnag-android
```

File: src/android/Bugsnag.cpp [hunk 2]
```diff
@@ -229,35 +222,40 @@ JNIEXPORT void JNICALL Java_com_bugsnag_android_BugsnagCocos2dxPlugin_configureN
...
+  _mapBreadcrumbType(env, bugsnag::Log, "LOG");
+  _mapBreadcrumbType(env, bugsnag::User, "USER");
+  _mapBreadcrumbType(env, bugsnag::State, "STATE");
+  _mapBreadcrumbType(env, bugsnag::Error, "ERROR");
+  _mapBreadcrumbType(env, bugsnag::Manual, "MANUAL");
+
   jclass bugsnag = env->FindClass("com/bugsnag/android/Bugsnag");
   bugsnag::BugsnagClass = (jclass)env->NewGlobalRef(bugsnag);
   env->DeleteLocalRef(bugsnag);
   bugsnag::BugsnagResumeSession = env->GetStaticMethodID(
```
Comment (lemnik) [inline src/android/Bugsnag.cpp:238]: While we didn't do it previously, should these not be GlobalRef'ed?

**Review by lemnik — APPROVED** [review]
LGTM with some minor comments / suggestions

File: src/android/Bugsnag.cpp (hunk 2 above)
Comment (kstenerud) [reply src/android/Bugsnag.cpp:238]: IDs stay valid so long as the class ref stays valid (which it does by virtue of the above `NewGlobalRef`). It's super annoying that some things are IDs and others are pointers that have to be made global...

File: src/android/Bugsnag.cpp [hunk 3]
```diff
@@ -229,35 +222,40 @@ JNIEXPORT void JNICALL Java_com_bugsnag_android_BugsnagCocos2dxPlugin_configureN
       bugsnag::HashMapClass, "put",
       "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
 
+  bugsnag::BreadcrumbTypeClass = env->FindClass("com/bugsnag/android/BreadcrumbType");
+  _mapBreadcrumbType(env, bugsnag::Navigation, "ERROR");
```
Comment (kattrali) [inline src/android/Bugsnag.cpp:226]:
```suggestion
  _mapBreadcrumbType(env, bugsnag::Navigation, "NAVIGATION");
```

Comment (kstenerud) [reply src/android/Bugsnag.cpp:226]: Damn... I'll open a new PR.

Comment (kattrali) [reply src/android/Bugsnag.cpp:226]: no worries, it took me a while to finish looking at this one!

**Pull request merged**